Bug fixes

PierreBizouard opened the pull request:
- Issues with python3, naming module
- Issues with tcp :0 binding with serve_in_thread, serve_in_process
- Issues with windows install on python2 (include_subdirs)
- Issues with socket checking in naming module

Idea of pizco-utils (qt dependant module for naming / system, service listing widgets and executables)

PierreBizouard committed:
added include subdirs
added main for windows two_servers.py example (due to absence of fork command)

PierreBizouard committed:
fixed bug on Linux (raspberrypi) not checking opened beacon port correctly

PierreBizouard committed:
missing %s causing log error

PierreBizouard committed:
bug using unexistent .contains instead of "in"

PierreBizouard committed:
bug in proxy creation timeout of remote naming service leading in zmq crash

PierreBizouard committed:
adding ignore ips options allowing certains ips (like masquerading routers) to be ignored

PierreBizouard committed:
Update naming with python2/3 compatibility
Solved few issues with serve_in_process and serve_in_thread, where instanciation comes after, and where the :0 binding needs to be communicated, also added environ lookup for naming ports (beacon udp/ beacon tcp)

PierreBizouard committed:
added specs for naming option in pizco-utils

hgrecco commented:
This looks very good. I will check it in depth and merge it on the weekend.

PierreBizouard commented:
Regarding the instantiate/inspect issue, the correction which might be a
workaround is calling inspect/instantiate/inspect. which is not completely
optimised.

I have started a "pizco-utils" project for distributing the naming system
as exe, providing a quick browsing with a pyqt dependency, for now only
placeholders and a basic spec is made. I let you know when the project will
be further. I have also in mind a cpp wrapper generator which could be part
of the project.

A+

2014-08-19 14:19 GMT+02:00 Hernan Grecco notifications@github.com:

> This looks very good. I will check it in depth and merge it on the weekend.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hgrecco/pizco/pull/21#issuecomment-52625001.

hgrecco commented:
looks good. By the way, have you look at http://nanomsg.org/
It seems to be a better zmq (but I have not tested yet). In particular, I like the fact that there is no context explicit.
The bad thing is that not so many people are using it right now. But if the people from IPython switch to it, we might do it as well as it will be available with all major python distros. Nothing to worry about know but maybe to keep in sight and write the pizco core in a way that switching is easy.

PierreBizouard commented:
thank you for nanomsg, i have also looked at the serialization mechanism
like http://msgpack.org/ which could also be an alternative to json.
zmq seems very wide spread, the patterns look similar so there may be a way
to find a common proxy for both, but premature optimization is the root of
all evil :)

2014-08-19 14:39 GMT+02:00 Hernan Grecco notifications@github.com:

> looks good. By the way, have you look at http://nanomsg.org/
> It seems to be a better zmq (but I have not tested yet). In particular, I
> like the fact that there is no context explicit.
> The bad thing is that not so many people are using it right now. But if
> the people from IPython switch to it, we might do it as well as it will be
> available with all major python distros. Nothing to worry about know but
> maybe to keep in sight and write the pizco core in a way that switching is
> easy.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hgrecco/pizco/pull/21#issuecomment-52626879.

hgrecco commented:
It seems that nanomsg is a reboot of zmq. I agree with you about premature optimization, I am more interested about the simplicity of the API than in the performance.
But I agree with you, I am very happy with pizco right now. Just something to keep in mind!

PierreBizouard committed:
massive bug fix
- naming now runs correctly
- removed time.sleep allways for wait loops on event
- added stop message on server
- added AgentManager bugfixes for loop closing
- proxystops know wait for end of the Proxy or Server
- test naming adds a few configurations

Added Naming server doc from Paul questions

PierreBizouard committed:
comment uncommented :)

PierreBizouard commented:
I have massively bugfixed stuff. In AgentManager for example. Hope it does not broke thing.

PierreBizouard committed:
wrong formatting for py2.6

PierreBizouard committed:
format issues

PierreBizouard committed:
bug correction in 2.6

PierreBizouard committed:
added function to change the ServicesWatcher method

PierreBizouard committed:
added ServiceWatcher method's check

PierreBizouard committed:
minor changes / add exception handling in beacon receiver

hgrecco commented:
Sorry for the long delay. I was going to merge this but the travis build is read. Any idea why?

PierreBizouard commented:
There are some issues left with the naming system. Maybe I should use less
memory in naming tests. I am very busy at work for the next 3 weeks, i
won't be vert helpfull. I have also added a blocking stop on stop function
maybe adding a timeout would help.
Le 19 oct. 2014 18:23, "Hernan Grecco" notifications@github.com a écrit :

> Sorry for the long delay. I was going to merge this but the travis build
> is read. Any idea why?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hgrecco/pizco/pull/21#issuecomment-59655070.

hgrecco commented:
It is fine. I apologize for not merging this sooner. Just ping back as soon as you have something to be merged. I commit to look at it promptly.

PierreBizouard committed:
added a bit of support to builtin types (usefull for pizcoutils), some minor issues with naming test : enforce server shut-down beetween different test configurations.

PierreBizouard committed:
bad print syntax

PierreBizouard committed:
test naming on another port > 1024

PierreBizouard committed:
updates to util due to auto signal functions

PierreBizouard committed:
force stop services beetween tests with infinite timeout

PierreBizouard committed:
suppression of eventqueue delayed command, start stop loop tries x5

PierreBizouard committed:
moved port from port 500 to 6500

PierreBizouard committed:
update in test naming, still one error to check

PierreBizouard committed:
removed old school print in naming test 
handle futures received to other proxies
handle pickle Server and Proxy and allways unpickle as Proxy (is it a good feature?)

PierreBizouard committed:
old school print removed

PierreBizouard commented:
Yeah it passes the tests!!! for 2.6 :/ i still have to check or remove comments

PierreBizouard committed:
py2.6 compat

PierreBizouard committed:
py2.6 compat

PierreBizouard committed:
old school print replacement

hgrecco commented:
It seems to be a problem with Python 2.6 but I have merged and we will fix it in my dev branch
Thanks a lot!
